astropy/vo/client/tests/test_vos_catalog.py

```diff
@@ -186,6 +186,9 @@ def test_write(self):
         outfile = os.path.join(self.outdir, 'test_1.json')
         db = vos_catalog.VOSDatabase.from_json(DB_FILE)
         db.to_json(outfile)
+        with pytest.raises(OSError) as exc:
+            db.to_json(outfile)
+        assert str(exc.value).endswith("test_1.json exists.")
```

Comment:
I went with the style given by the [pytest](http://doc.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions) documentation, which puts the assertion outside the with block.
In fact, most (if not all) of the astropy test code follows this style.